src/framework/ui/tabNavigator/bottomNavigatorTab.component.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import {
+  View,
+  ViewProps,
+  Text,
+  StyleSheet,
+  ImageSourcePropType,
+  TextProps,
+  ImageProps,
+  Image,
+} from 'react-native';
+import {
+  StyledComponentProps,
+  StyleType,
+} from '@kitten/theme';
+
+interface BottomNavigatorTabProps {
+  appearance?: string;
+  title?: string;
+  isSelected?: boolean;
+  getIconSource?: (isSelected: boolean) => ImageSourcePropType;
+}
+
+export type Props = BottomNavigatorTabProps & StyledComponentProps & ViewProps;
+
+export class BottomNavigatorTab extends React.Component<Props> {
+
+  private hasText(): boolean {
+    return this.props.title && this.props.title.length !== 0;
+  }
+
+  private getComponentStyle = (source: StyleType): StyleType => ({
+    icon: {
+      width: source['icon.width'],
```

Comment:
is it ok to have such selectors?

Reply:
@malashkevich this can be refactored with token groups like we did this with `text`
For now for huge components which contain lots of children this is one of the way we can achieve this